src/events/channels/transactionsChannel.ts

```diff
@@ -0,0 +1,180 @@
+          : 0
+
+        let jobStatus = await prisma.cronJobStatus.findFirst({
+          where: { jobName: JobName.TRANSACTIONS }
+        })
+
+        if (!jobStatus) {
+          jobStatus = await prisma.cronJobStatus.create({
+            data: {
+              jobName: JobName.TRANSACTIONS,
+              state: JSON.stringify({
+                lastHeight: currentHeight
+              })
+            }
+          })
+        }
+
+        const lastCheckHeight = (
+          JSON.parse(jobStatus.state as string) as { lastHeight: number }
+        ).lastHeight
+
+        // Determine fetch interval
+        let heightToFetch = lastCheckHeight + config.app.heightSkipPerHeight
+
+        if (heightToFetch > currentHeight) {
+          this.isLocked = false
+          return
+        }
+
+        if (currentHeight - heightToFetch > 1) {
+          // If the gap is more than 1 block, fetch transactions in chunks of 1000 blocks
+          heightToFetch = lastCheckHeight + 1000
+
+          if (currentHeight - heightToFetch < 0) {
+            // if the gap reaches currenHeight and more, than set it to currentHeight
+            heightToFetch = currentHeight
+          }
+        }
+
+        const txs = await adamantClient.getTransactions({
+          fromHeight: lastCheckHeight + 1,
+          and: {
+            toHeight: heightToFetch
+          },
+          returnAsset: 1
+        })
+
+        if (!txs.success) {
+          this.isLocked = false
+          return
+        }
+
+        txs.transactions.forEach((tx) => {
+          if (tx.height < currentHeight - config.notify.latestHeightToNotify) {
+            // skip if transaction is too old
+            return
+          }
+
+          this.handleTransaction(tx)
+        })
+
+        await prisma.cronJobStatus.update({
+          where: { jobName: JobName.TRANSACTIONS },
+          data: {
+            state: JSON.stringify({
+              lastHeight: heightToFetch
+            })
+          }
+        })
+      } catch (error) {
+        logger.error(error, 'Error while running transactions channel job')
+      } finally {
+        this.isLocked = false
+      }
+    })
+
+    this.job.start()
+  }
+
+  handleTransaction(tx: AnyTransaction) {
+    if (this.processedTxs[tx.id]) {
+      delete this.processedTxs[tx.id] // removing from cache because we got tx again from rest api or socket
+      return
+    }
+
+    this.processedTxs[tx.id] = tx
+
+    if (isSignalTx(tx)) {
+      logger.info(
+        `Got signal transaction to (un)subscribe to notifications, txId: ${tx.id}, processing...`
+      )
+      this.emit('newSignalMessage', tx as ChatMessageTransaction)
+    } else if (isTxToNotify(tx)) {
+      this.emit('newMessage', tx)
+    }
+  }
+
+  destroy() {
+    if (this.job) {
+      this.job.stop()
+      logger.info('Stopped TransactionsJob job')
+    }
+
+    if (this.adamantSocket) {
+      this.adamantSocket.off(this.handleTransaction)
```

Comment:
This will not work because you used .bind earlier that creates a new function reference. Use arrow function declaration and remove bind to solve this.

Example: handleTransaction = () => {}
adamantClient.on(this.handleTransaction)
adamantClient.off(this.handleTransaction)

Reply:
@bludnic fixed, added handleTransactionRef as private readonly property